File: PadelClub.Services/Database/MatchParticipant.cs
using System;

namespace PadelClub.Services.Database
{
    public class MatchParticipant
    {
        public int Id { get; set; }
        public int MatchId { get; set; }
        public int UserId { get; set; }
        public int TeamNumber { get; set; } // 1 or 2
        public string Role { get; set; } = "Player"; // Player, Captain
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        public virtual Match Match { get; set; } = null!;
        public virtual User User { get; set; } = null!;
    }
}

File: UI/padelclub_desktop/lib/features/match_participants/domain/entities/match_participant.dart
class MatchParticipant {
  final int? id;
  final int matchId;
  final int userId;
  final int teamNumber;
  final String role;
  final DateTime createdAt;

  MatchParticipant({
    this.id,
    required this.matchId,
    required this.userId,
    required this.teamNumber,
    required this.role,
    required this.createdAt,
  });
}
